java/src/hdf/hdf5lib/HDFArray.java

```diff
@@ -169,10 +169,6 @@ else if (ArrayDescriptor.className
                                 .equals("java.lang.Integer")) {
                             therow = IntegerToByte((Integer[]) _theArray);
                         }
-                        else if (ArrayDescriptor.className
```

Comment:
Why was this removed?

Reply:
The changes in this file were to correct Java compile failures on one of the THG machines (sorry, can’t remember which one, maybe @lrknox remembers), which were failing with a complaint about a switch using a string variable.

Reply:
I think this file and the changes need to be extracted to a separate issue for further investigation.

Reply:
Created PR just for this file